plugins/reindex/src/main/java/org/elasticsearch/plugin/reindex/AbstractBulkByScrollRequest.java

```diff
@@ -72,6 +74,17 @@
      */
     private WriteConsistencyLevel consistency = WriteConsistencyLevel.DEFAULT;
 
+    /**
+     * Initial delay after a rejection before retrying request. Defaults to 50ms. With the default maxRetries to total time to retry
+     * rejects is about 25 seconds per rejection.
+     */
+    private TimeValue retryBackoffInitialTime = timeValueMillis(50);
+
+    /**
+     * Total number of retries attempted for rejections. Defaults to 10.
+     */
+    private int maxRetries = 10;
```

Comment:
Chiming in for Nik: In the way it is exposed here I think this won't work because internally it uses an exponential backoff anyway. The API would need to allow you to set a different backoff policy explicitly. Then you could just define your own backoff policy. Note that we have a few standard implementations but so far none that retries forever. (see also [BackoffPolicy.java in master](https://github.com/elastic/elasticsearch/blob/master/core/src/main/java/org/elasticsearch/action/bulk/BackoffPolicy.java))